src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1871,12 +1860,23 @@ private BetterResult BetterConversionFromExpression(BoundExpression node, TypeSy
                         return conv2.Kind == ConversionKind.Identity ? BetterResult.Right : BetterResult.Neither;
                     }
                 }
-                else if (refKind1 == RefKind.Ref)
+                else if (p1.RefKind == RefKind.Ref)
                 {
                     return BetterResult.Neither;
                 }
             }
 
+            // If an ambiguity exists between 'by-val' and 'in' parameter, choose the 'by-val' one.
+            // Except if it was params. Subsequent betterness analysis will always prefer the non-params one.
+            if (!p1.IsParams && p1.RefKind == RefKind.None && p2.RefKind == RefKind.In)
```

Comment:
This is confusing. 

If we have
```C#
M1(params int[] arr) . . . 
```

and 
```C#
M1(in int[] arr) ...
```

And I call with ```M1( new int[] {1,2})```  -  i.e. first is applicable in normal (not expanded) form
I would expect the byval method to be selected, regardless if it was declared params or not.  The comment seems to say the opposite.

We need some tests for this scenario - with params overload matching in normal or expanded form. (in expanded form I expect it to not be selected)

Reply:
@VSadov We already have a test that tests params expansion with in: `PassingInArgumentsOverloadedOnInParams`.
For the array variation, we are still reporting ambiguity (I prefer this over selecting the by-val one, as it can break the caller). Will add another test for that variation and explain in the comment.